src/allocators/OnChainAllocator.sol

```diff
@@ -579,25 +622,31 @@ contract OnChainAllocator is IOnChainAllocator, Utility {
     }
 
     function _getAndUpdateNonce(address calling, address sponsor) internal returns (uint256 nonce) {
+        // Create an on chain nonce by combining the command, the calling address and the next free nonce.
+        // Updates the free nonce pointer.
+        bytes1 onChainNonceCommand = AL.ON_CHAIN_NONCE;
         assembly ("memory-safe") {
             sponsor := mul(sponsor, iszero(calling))
             mstore(0x00, sponsor)
             mstore(0x20, nonces.slot)
             let nonceSlot := keccak256(0x00, 0x40)
-            let nonce96 := sload(nonceSlot)
-            nonce := or(shl(96, sponsor), add(nonce96, 1))
-            sstore(nonceSlot, add(nonce96, 1))
+            let nonce88 := sload(nonceSlot)
+            nonce := or(onChainNonceCommand, or(shl(88, sponsor), add(nonce88, 1)))
```

Comment:
minor thing but we could do the add right when we do the sload and save an operation over needing to do it twice